addons/hw_drivers/controllers/driver.py

```diff
@@ -560,19 +560,26 @@ def run(self):
 
 class SocketManager(Thread):
 
+    def __init__(self):
+        super(SocketManager, self).__init__()
+        self.open_socket(9000)
+
+    def open_socket(self, port):
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.sock.bind(('', port))
+        self.sock.listen(1)
+        self.sock.settimeout(15.0)
+
     def run(self):
         while True:
             try:
-                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-                sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-                sock.bind(('', 9000))
-                sock.listen(1)
-                dev, addr = sock.accept()
+                dev, addr = self.sock.accept()
```

Comment:
In the connection issue experienced by the customers, I've observed the Ingenico terminal first having a working connection with the IoT box and I could see transactions happening and keepalive messages going back and forth. Next the Ingenico terminal created a second connection to the box without closing the first. It sends a hello request on the second connection, and after that communication between both parties stops.

If a second connection comes in from the same ip here, `addr[0]` is going to be the same for both connections. This means the second connection gets ignored, because the key is already in `socket_devices`, and it won't be closed either because `dev` still points to it. So we end up in a situation where the terminal is waiting for a response on the second connection, which won't happen. Meanwhile, the IoT box is waiting for new messages from the terminal on the first connection in the IngenicoDriver thread, which also doesn't happen. Hence a deadlock.

It's not clear at this point why the Ingenico terminal is creating a second connection, it needs to be investigated further.

There's a couple of options on how we could deal with this. We could replace the old connection with the new one, or we could disconnect both so that the terminal has to start over. Any thoughts?

This logic is also present in the current stable branch: https://github.com/odoo/odoo/blob/6e7c0be2733a3fdebd56599bb75fc199ae7dd169/addons/hw_drivers/controllers/driver.py#L571 , which would explain why the issue remained despite this PR, and why reverting https://github.com/odoo/enterprise/commit/e09fb254e4dc3218bedbf769c2d57613c123a7ae didn't help.